MSAL/src/native_auth/public/MSALNativeAuthPublicClientApplication+Internal.swift

```diff
@@ -49,72 +52,48 @@ extension MSALNativeAuthPublicClientApplication {
             context: context
         )
 
-        return await controller.signUpStartPassword(parameters: parameters)
-    }
-
-    func signUpInternal(
-        username: String,
-        attributes: [String: Any]?,
-        correlationId: UUID?
-    ) async -> MSALNativeAuthSignUpControlling.SignUpStartCodeControllerResponse {
-        guard inputValidator.isInputValid(username) else {
-            return .init(.error(SignUpStartError(type: .invalidUsername)))
+        if password != nil {
+            return await controller.signUpStartPassword(parameters: parameters)
+        } else {
+            return await controller.signUpStartCode(parameters: parameters)
         }
-
-        let controller = controllerFactory.makeSignUpController()
-        let context = MSALNativeAuthRequestContext(correlationId: correlationId)
-
-        let parameters = MSALNativeAuthSignUpStartRequestProviderParameters(
-            username: username,
-            password: nil,
-            attributes: attributes,
-            context: context
-        )
-
-        return await controller.signUpStartCode(parameters: parameters)
     }
 
-    func signInUsingPasswordInternal(
+    func signInInternal(
         username: String,
-        password: String,
+        password: String?,
         scopes: [String]?,
         correlationId: UUID?
-    ) async -> MSALNativeAuthSignInControlling.SignInPasswordControllerResponse {
+    ) async -> MSALNativeAuthSignInControlling.SignInControllerResponse {
         guard inputValidator.isInputValid(username) else {
-            return .init(.error(SignInPasswordStartError(type: .invalidUsername)))
+            return .init(.error(SignInStartError(type: .invalidUsername)))
         }
 
-        guard inputValidator.isInputValid(password) else {
-            return .init(.error(SignInPasswordStartError(type: .invalidCredentials)))
+        if let password = password {
+            guard inputValidator.isInputValid(password) else {
+                return .init(.error(SignInStartError(type: .invalidCredentials)))
+            }
         }
 
         let controller = controllerFactory.makeSignInController()
 
-        let params = MSALNativeAuthSignInWithPasswordParameters(
-            username: username,
-            password: password,
-            context: MSALNativeAuthRequestContext(correlationId: correlationId),
-            scopes: scopes
-        )
-
-        return await controller.signIn(params: params)
-    }
+        if let password = password {
```

Comment:
Do we still need this distinction? What if we have only one signIn controller method?